### PR TITLE
Fix empty General Collateral on Review page

### PR DESCRIPTION
**Pull request by dimak1**
*Issue #:* /bcgov/entity#12430

Deleting all text from General Collateral description (wysiwyg) does not clear HTML styling tags, which in turn displays empty Collateral Summary

*Description of changes:*
- Fix the issue by stripping out HTML from the GC description and checking for valid text
- Add tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/collateral/Collateral.vue [hunk 1]
```diff
@@ -192,6 +193,18 @@ export default defineComponent({
...
+      if (getAddCollateral.value.generalCollateral[0]?.description?.replace(/(<([^>]+)>)/ig, '').length === 0) {
+        setGeneralCollateral([])
+        // clear collateral step check mark if there are no vehicles
+        // (this resets check mark that was set by general collateral description)
+        if (getAddCollateral.value.vehicleCollateral.length === 0) {
+          getAddCollateral.value.valid = false
+        }
+      }
+    })
+
```
Comment (cameron-eyds) [inline ppr-ui/src/components/collateral/Collateral.vue:207]: Very cool, just curious, perhaps you already tried this as well but what if we made this conditional a computed property(so a reactive boolean in this case) and we set up a watcher on that property to clear the store?

Also, with regards to the `getAddCollateral.value.valid` is this computed currently based on have some char length or set manually by ui?

Comment (cameron-eyds) [reply ppr-ui/src/components/collateral/Collateral.vue:207]: I'm just a little wary of using unmounted lifecycle hook in a SPA, although i'm sure it's probably fine here..

Comment (dimak1) [reply ppr-ui/src/components/collateral/Collateral.vue:207]: I tried to add a condition [here](https://github.com/bcgov/ppr/blob/cf72314e0ad3a3278ce8f7dc77bb72d26a12f3cd/ppr-ui/src/components/collateral/generalCollateral/GenColEdit.vue#L146) in a watcher to check if the stripped value had any leftover text but it was lagging (delay on each keystroke). So then I wanted to check only once as the user navigates away from that route. 
`Valid` is set manually.

Comment (cameron-eyds) [reply ppr-ui/src/components/collateral/Collateral.vue:207]: Ok, fair enough, thanks!

**Review by cameron-eyds — APPROVED** [review]
Looks good 👍

**Review by doug-lovett — APPROVED** [review]

**Pull request merged**